aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -252,6 +252,17 @@ static void clamp_min_kernel(TensorIterator& iter, Scalar min_scalar) {
   });
 }
 
+static void cauchy_kernel(TensorIterator& iter, double median, double sigma, Generator* gen) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "cauchy_cpu", [&]() {
+    CPUGenerator* generator = get_generator_or_default<CPUGenerator>(gen, detail::getDefaultCPUGenerator());
+    std::lock_guard<std::mutex> lock(generator->mutex_);
+    cpu_serial_kernel(iter, [median, sigma, generator]() -> scalar_t {
+      at::cauchy_distribution<double> cauchy(median, sigma);
```

Comment:
As written here you're regenerating `cauchy_distribution` every loop iteration and I'm not sure if it gets hoisted out. Well, the benchmark seems to think it's fine so maybe it is hoisted out.